lib/clustering/reclusterer.rb

```diff
@@ -89,29 +92,20 @@ def cluster_components
     end
 
     def recluster_components
-      @cluster.ocn_resolutions
-        .each {|r| ClusterOCNResolution.new(r.dup).cluster.save }
-      recluster_ht_items
-      recluster_holdings
-      recluster_commitments
-    end
-
-    def recluster_batch(clusterables, sort_field, clusterer)
-      clusterables.sort_by(&sort_field)
-        .chunk_while {|item1, item2| item1.batch_with?(item2) }
-        .each {|batch| clusterer.new(*batch.map(&:dup)).cluster.save }
-    end
-
-    def recluster_ht_items
-      recluster_batch(@cluster.ht_items, :ocns, ClusterHtItem)
-    end
-
-    def recluster_holdings
-      recluster_batch(@cluster.holdings, :ocn, ClusterHolding)
+      graph.subgraphs.each do |ocn_set|
+        new_cluster = Cluster.where(ocns: { "$in": ocn_set }).first || Cluster.new(ocns: ocn_set)
+        move_components(new_cluster.ht_items, @cluster.ht_items.where(ocns: { "$in": ocn_set }))
```

Comment:
That hits the database 5 times per new_cluster when we can get away with doing one save. Does it matter? I dunno.

Reply:
I had thought we found that using `<<` either hit mongo or was otherwise not performant in some kind of `n^2` fashion, but in reviewing when we started using the `$push` operator it was here: 
https://github.com/hathitrust/holdings-backend/commit/45c0901fe726d292f2807648feb6b97dfc3d44bb

with the purpose of detecting if updates failed & retrying them. 

That shouldn't be an issue in creating a new cluster, so I think this is OK as is.